sdk/core/Azure.Core/tests/RetriableStreamTests.cs

```diff
@@ -344,6 +344,48 @@ public async Task FlushDoesntThrow()
             await reliableStream.FlushAsync();
         }
 
+        [Test]
+        public async Task CanThrowOnRead()
```

Comment:
should we also have some tests that just assert that both the `onRead` and `onDispose` Actions get called as expected?

Reply:
Good call out, will do - thanks!

Reply:
Now moot, per new approach.